src/azul/indexer/__init__.py

```diff
@@ -83,10 +83,43 @@ class SimpleSourceSpec(SourceSpec['SimpleSourceSpec']):
 
     @classmethod
     def parse(cls, spec: str) -> 'SimpleSourceSpec':
-        return cls(name=spec)
+        """
+        >>> s = SimpleSourceSpec.parse('https://foo.edu:12')
+        >>> s
+        SimpleSourceSpec(prefix='12', name='https://foo.edu')
+
+        >>> s = SimpleSourceSpec.parse('foo:bar/baz:12')
+        >>> s
+        SimpleSourceSpec(prefix='12', name='foo:bar/baz')
+
+        >>> s = SimpleSourceSpec.parse('foo')
+        >>> s
+        SimpleSourceSpec(prefix='', name='foo')
+        """
+        # FIXME: Move parsing of prefix to SourceSpec
+        #        https://github.com/DataBiosphere/azul/issues/3073
+        if ':' in spec:
```

Comment:
The double lookup can be eliminated:

![image](https://user-images.githubusercontent.com/5143256/119746340-26541e80-be45-11eb-9e87-2641d6d9db0f.png)

Reply:
Anyways, the colon shouldn't be optional. But the condition is violating the symmetry invariant I mentioned in our convo.